Sources/ProjectDescription/Headers.swift

```diff
@@ -82,4 +88,48 @@ public struct Headers: Codable, Equatable {
             exclusionRule: exclusionRule
         )
     }
+
+    public static func headers(from list: FileList,
+                               umbrella: Path,
+                               private: FileList? = nil,
+                               allOthersAsProject: Bool) -> Headers
+    {
+        return .init(
+            public: list,
+            umbrellaHeader: umbrella,
+            private: `private`,
+            project: allOthersAsProject ? list : nil,
+            exclusionRule: .projectExcludesPrivateAndPublic
+        )
+    }
+
+    /// all left found headers from the list
+    /// (after public/private scopes loading)
+    /// will be added with `project` visibility
+    public static func allHeaders(from list: FileList,
+                                  umbrella: Path,
+                                  private: FileList? = nil) -> Headers
+    {
+        return headers(
+            from: list,
+            umbrella: umbrella,
+            private: `private`,
+            allOthersAsProject: true
+        )
+    }
+
+    /// all left found headers from the list
+    /// (after public/private scopes loading)
+    /// will be skipped
+    public static func onlyHeaders(from list: FileList,
```

Comment:
Same here, what's the use case for this method?

Reply:
And here: "I want only headers from list, which exists in umbrella list and all headers from `private` list, all other should be ignored (e.g. they are for unit-tests or/and other target)